proposals/0230-spp-resource-management-for-android.md

```diff
@@ -36,175 +35,139 @@ and define the setter:
+
+### When user clicks on notification popup
+
+In previous code snippet (Code-1), SdlNotificationActivity is specified for ContentIntent. The activity is shown when user clicks on the notification popup.
+
+The sample UX of SdlNotificationActivity looks like below:
+
+![SdlNotificationActivity UX](../assets/proposals/0230-spp-resource-management-for-android/spp_error_activity.png)
+
+All strings used for this error UX is defined in strings.xml like below:
 
-    /**
-     * getLegacyModeAllowed (getter)
-     * @return
-     */
-    public boolean getLegacyModeAllowed() {
-        return this.legacyModeAllowed;
-    }
-}
-```
-```enterLegacyMode``` checks to see ```transportListener.onLegacyModeEnabled```, and we can check the flag in ```onLegacyModeEnabled``` something like below:
 ```java
-in SdlProtocol:
-
-        public boolean onLegacyModeEnabled(String info) {
-            //Await a connection from the legacy transport
-            if(transportConfig.getLegacyModeAllowed() && requestedPrimaryTransports!= null && requestedPrimaryTransports.contains(TransportType.BLUETOOTH)
-                    && !transportConfig.requiresHighBandwidth()){
-                Log.d(TAG, "Entering legacy mode; creating new protocol instance");
-                reset();
-                return true;
-            }else{
-                Log.d(TAG, "legacy mode is not allowed");
-                return false;
-            }
-        }
+    <string name="spp_out_of_resource">Bluetooth channel is out of resource</string>
+    <string name="notification_title">SmartDeviceLink</string>
+    <string name="spp_out_of_resource_message">There are too many bluetooth apps running on your device. Please close them and try to re-connect</string>
+    <string name="button_ok">OK</string>
+    <string name="sdl_error_notification_channel_name">SDL Error</string>
 ```
 
+The strings above are TBD (requires review).
+
+### Localization of SPP error UX
+
+Because all strings used for error UX is defined in strings.xml, the standard localization process for Android can be applied; i.e. adding language specific strings.xml per language.
+All language enums of [Mobile_API](https://github.com/smartdevicelink/sdl_core/blob/master/src/components/interfaces/MOBILE_API.xml#L167) are assumed as the supported languages. They are:
+
+ar_SA, cs_CZ, da_DK, de_DE, el_GR, en_AU, en_GB, en_IN, en_SA, en_US, es_ES, es_MX, fi_FI, fr_CA, fr_FR, he_IL, hi_IN, hu_HU, id_ID, it_IT, ja_JP, ko_KR, ms_MY, no_NO, pl_PL, pt_BR, pt_PT, ro_RO, ru_RU, sk_SK, sv_SE, th_TH, tr_TR, uk_UA, vi_VN, zh_TW, zh_CN
+
+The default language would be en_US.
+
 ## Potential downsides
 
-It is not the downside, but the application should be responsible for implementing the notification listener when SPP resource runs out. The implementation guide should be updated accordingly.
+- We need to think about who is responsible for the localization process, which includes management of string resources.
```

Comment:
It appears a revision requested by the Steering Committee was:
>Author to propose who they believe should be responsible for obtaining and maintaining translations of this string, setting the precedent for future similar cases.

I believe this still needs to be added.  Please let me know if I'm misunderstanding.

Reply:
You're right. I have removed the open question from potential downsides section, and then added proposed localization process in "Localization of SPP error UX" section. Hope this meets the request by the Steering Committee.